PR: https://github.com/odoo/odoo/pull/

From: ea66192276921be1bafe73255fa77c55f3e10ac7
From: Haresh Shyara
Date: 2017-08-09 08:03:08

Breaking data model changes scores: del:8 + add:0, change matches:
-    _inherit = 'res.config.settings'
-    company_id = fields.Many2one('res.company', string='Company', required=True,
-    module_crm = fields.Boolean("CRM")
-    module_sale = fields.Boolean("Sales")
-    module_website_sign = fields.Boolean("eSign")
-    module_helpdesk = fields.Boolean("Helpdesk")
-    module_sale_subscription = fields.Boolean("Subscriptions")
-    module_account_accountant = fields.Boolean("Accounting")

Total Changes: 19

[REF] crm,sale: Split the applications into 2 separate apps

Purpose
=======

You install CRM but you don't see CRM icon (except in English, as long as sales is not installed) -> confusing
The dashboard with salesteam is not obvious & not useful for simple users: what are those channel boxes about? why don't I see all my activities?, etc.

Specifications
==============

Split CRM & Sales + land in business views straight away

Default Entry Views
~~~~~~~~~~~~~~~~~~~

* New default view for CRM: Pipeline kanban
* New default view for Sales: Quotation list

Dashboards
~~~~~~~~~~

* Move sales teams dashboard to Reporting: Sales Channels.
* Remove the upper part of the sales team dashboard

Menuitems Structure
~~~~~~~~~~~~~~~~~~~

* CRM menu

Pipeline:
	- Leads (optional)
	- Pipeline (default menu, default filter "My Pipeline")
	- Next Activities -> to remove (now there is an icon in top bar for next activities)
	- Quotations (If sale_management installed)
Customers
Phone Calls (if voip, if not yet removed because there is a BE task to use next activities instead)
Leads Management:
	- Scoring Rules
	- Leads Assignation
	- Team Assignation
Reporting:
	- Leads
	- Scoring page views
	- Opp. Assignement
	- Patnerships
	- Pipeline
	- Activities
	- Phonecalls
	- Sales Channels
Configuration:
	- Settings
	- Sales Channels
	- Activity Types
	- Leads & Opportunities:
		- Lead Tags
		- Lost Reasons
	- Resellers
		- Partner Level
		- Partner Activations
* Sales menu:

Orders:
	- Quotations
	- Orders
	- Customers
Invoicing:
	- Orders to Invoice
	- Orders to Upsell
Catalog:
	- Products
	- Promotion Programs
	- Coupon Programs
Reporting:
	- Sales
	- Sales Channels
	- All channels sales orders
Configuration:
	- Settings
	- Sales Channels
	- Sales Orders:
		- Quotation templates
		- Payment acquirers
		- Delivery Methods

Settings
~~~~~~~~

* Split settings form
* Remove recommended apps section
* Remove Timesheets section
* Remove Inventory Management and move shipping connectors to Shipping section
* Remove Integrations section
	- Move Docsaway option to "Quotations & Orders", right after proforma
	- Autocomplete and asterix: move to CRM section
* Restructure and rename sections this way:
	- Product Catalog
	- Pricing
	- Quotations & Orders
	- Shipping
	- Invoicing
	- eBay (if installed)
* Remove all the "Save this page and come back here" when checking a box that installs a new module

* restructure a bit the CRM settings, with 3 sections:

Pipeline
	- Leads
Contacts
	- Phone Validation
	- Customer Autocomplete
Integrations
	- Google Calendar / Synchronize your calendar with Google Calendar (copy from general settings but don't show message "Save this page and come back here...".
	- Asterisk (VOIP)

================================= pseudo patch: =================================

--- a/addons/sales_team/models/__init__.py
+++ b/addons/sales_team/models/__init__.py
@@ -4,4 +4,3 @@
 from . import crm_team
 from . import res_partner
 from . import res_users
-from . import sale_config_settings

--- a/addons/sales_team/models/sale_config_settings.py
+++ b/None
@@ -1,18 +0,0 @@
-# -*- coding: utf-8 -*-
-# Part of Odoo. See LICENSE file for full copyright and licensing details.
-
-from odoo import fields, models
-
-
-class SaleConfigSettings(models.TransientModel):
-    _name = 'sale.config.settings'
-    _inherit = 'res.config.settings'
-
-    company_id = fields.Many2one('res.company', string='Company', required=True,
-        default=lambda self: self.env.user.company_id)
-    module_crm = fields.Boolean("CRM")
-    module_sale = fields.Boolean("Sales")
-    module_website_sign = fields.Boolean("eSign")
-    module_helpdesk = fields.Boolean("Helpdesk")
-    module_sale_subscription = fields.Boolean("Subscriptions")
-    module_account_accountant = fields.Boolean("Accounting")
